Cards/CardBase.gd

```diff
@@ -35,12 +35,41 @@ func _apply_all_effects(target: Entity) -> void:
 	for effect_data: EffectData in card_effects_data:
 		effect_data.apply_effect_data(target)
 
+func _apply_effect(target : Entity, effect: EffectData):
+	effect.apply_effect_data(target)
 
 func can_play_card(caster: Entity, target: Entity) -> bool:
 	return caster.get_party_component().can_play_on_entity(application_type, target)
 
 
-func on_card_play(caster: Entity, target: Entity) -> void:
-	_apply_all_effects(target)
+func on_card_play(caster: Entity, targets: Array[Entity]) -> void:
+	#Split up targeted attacks and all attacks
+	var target_effects: Array[EffectData] = []
+	var all_effects: Array[EffectData] = []
+	
+	for effect_data: EffectData in card_effects_data:
+		if(effect_data.target_type == Enums.TargetType.SINGLE_TARGET):
+			target_effects.push_front(effect_data)
```

Comment:
why use `push_front` instead of `append` ? this will make effects play in reverse
for example if you have a card that says "Does A, then B, then C", when we read it the list will be `[A]` then we push B to the front it becomes `[B,A]` then `[C,B,A]`

Reply:
I don't know why I did it like that either, I'll change it